### PR TITLE
Add documentation about REST APIs

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
Adds documentation about the REST APIs exposed by this plugin.

### CODE REVIEW

**Comment by AbhyudayaSharma** [comment]
I tried using Swagger.json and have created the configuration for one of the APIs  https://app.swaggerhub.com/apis-docs/abhyudaya/folder-auth/1.0.0

Is writing the YAML manually the correct way to go?

**Comment by oleg-nenashev** [comment]
I think adding manual YAML is fine for now, because it might help users. We
can always rework it later

File: docs/rest-api.adoc [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+Deletes an agent role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/deleteGlobalRole`. The parameter
+`roleName` is required.
+
+[source,bash]
+----
+curl -X POST -d 'roleName=foo' http://localhost:8080/folder-auth/deleteAgentRole
+----
+
+== Logging in to Jenkins
```
Comment (oleg-nenashev) [inline docs/rest-api.adoc]: It is better to use API tokens for credentials.
Also, a token will be needed on instances with CSRF protection
https://wiki.jenkins.io/display/JENKINS/Remote+access+AP

File: docs/rest-api.adoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+administrator permissions. See the example below for viewing the home page:
+
+[source,bash]
+----
+curl -X GET -u $USERNAME:$TOKEN http://localhost:8080/
+----
+
+The API token can be obtained by clicking on your logged in user on the top-right
+of your Jenkins Home page and then clicking the 'Configure' button in the side bar.
+For more information about authentication, please see https://wiki.jenkins.io/display/JENKINS/Remote+access+API
```
Comment (AbhyudayaSharma) [inline docs/rest-api.adoc]: Is this better?

**Pull request merged**